### PR TITLE
Save the fact that content is an ical content

### PR DESCRIPTION
**Pull request by axi**
The parser does already the job, so we can use it to "validate" it's really an ical content.

### CODE REVIEW

File: src/ICal/ICal.php [hunk 1]
```diff
@@ -1351,7 +1359,7 @@ protected function processRecurrences()
             // Separate the RRULE stanzas, and explode the values that are lists.
             $rrules = array();
             foreach (array_filter(explode(';', $anEvent['RRULE'])) as $s) {
-                list($k, $v) = explode('=', $s);
+                [$k, $v] = explode('=', $s);
```
Comment (u01jmg3) [inline src/ICal/ICal.php:1362]: Please revert this unmentioned change

**Review by u01jmg3 — CHANGES_REQUESTED** [review]

File: src/ICal/ICal.php (hunk 1 above)
Comment (axi) [reply src/ICal/ICal.php:1362]: Sure, I missed this one

**Comment by axi** [comment]
Hey @u01jmg3, could you check this PR ? Thanks

**Comment by axi** [comment]
just thinking about this, I'm using the parser in a loop and before passing it a new url to parse, I have to 
```
                // Reset from potential previous iteration
                $ical->cal = [];
                $ical->isIcalContent = false;
```
Should'nt that be in a new `reset()` method ?
I can update the PR (or create a new one probably)

**Comment by u01jmg3** [comment]
I've sat on this one for a bit and I don't think knowing that a file starts with `BEGIN:VCALENDAR` is enough to say you've validated it as iCal content.

I think if you need this in your code you should track it yourself.

**Comment by axi** [comment]
Ok. For reference,  this would probably be the minimum for validating a ical content: https://stackoverflow.com/a/67698638/2454746

**Pull request closed without merging**